lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -336,6 +336,32 @@ defmodule ExUnit.AssertionsTest do
       "Expected exception SyntaxError but got FunctionClauseError (no function clause matching in :lists.flatten/1)" = error.message
   end
 
+  test "assert raise comparing messages (for equality)" do
+    assert_raise RuntimeError, "foo", fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  \"foo\"" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
+  end
+
+  test "assert raise comparing messages (with a regex)" do
+    assert_raise RuntimeError, ~r/ba[zk]/, fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      "Wrong message for RuntimeError" <>
+      "\nExpected:" <>
+      "\n  ~r/ba[zk]/" <>
+      "\nGot:" <>
+      "\n  \"bar\"" = error.message
```

Comment:
I followed [this](https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/test/ex_unit/assertions_test.exs#L157-L161) lines in the same test file. That said, I'm seeing both styles are used [elsewhere](https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/test/ex_unit/assertions_test.exs#L122-L126) in that file. Anyways, I wouldn't worry so much about such a "style" issue. Another commit just to change this is history pollution IMO. What do you think?